server/index.js

```diff
@@ -63,6 +63,105 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!req.session.cartId) {
+    return res.json([]);
+  } else {
+    const sql = `
+      select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+       from "cartItems" as "c"
+       join "products" as "p" using ("productId")
+      where "c"."cartId" = $1
+    `;
+    const params = [cartId];
+    db.query(sql, params)
+      .then(result => {
+        return res.json(result.rows);
+      })
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  const params = [productId];
+
+  const sql = `
+    select "price"
+    from "products"
+    where "productId" = $1
+  `;
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return next(new ClientError('productId must be a positive integer', 404));
+  }
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows) {
+        return next(new ClientError('No results available at this moment', 400));
+      } else {
```

Comment:
You need to `throw` here instead of `return next()`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw